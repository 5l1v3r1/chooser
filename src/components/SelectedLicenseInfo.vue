<template>
    <div id="SelectedLicenseInfo">
        <h4>{{ value.fullName }}</h4>
        <div>
            <ul id="license-list">
                <li v-for="item in iconsList"
                    :key="item">
                    <LicenseIconography
                        :icon-list="[item]"/>
                    <span class="readable-string">
                        {{ readableString(item)}}
                    </span>
                </li>
            </ul>
        </div>
        <LicenseDescription
            :selectedLicense="value.shortName"/>
    </div>
</template>
<script>
import LicenseDescription from './LicenseDescription'
import LicenseIconography from './LicenseIconography'
import licenseUrl from '@/utils/licenseUrl'

export default {
    name: 'SelectedLicenseInfo',
    props: ['value'],
    components: {
        LicenseDescription,
        LicenseIconography
    },
    methods: {
        readableString(shortString) {
            const readableStrings = {
                cc0: 'Public Domain',
                by: 'The creator must be credited',
                nc: 'Commercial use not permitted',
                nd: 'No derivatives or modification permitted',
                sa: 'Share allowed only on the same conditions'
            }

            return readableStrings[shortString] ? readableStrings[shortString] : 'Public Domain'
        }
    },
    computed: {
        iconsList() {
            const slicedName = this.$props.value.shortName
                .slice(3, this.$props.value.shortName.length - 4)
            return slicedName.toLowerCase().split('-')
        }
    },
    mounted: function() {
        console.log('SelectedLicenseInfo mounted, licenseUrl: ', this.$refs, licenseUrl)
        // this.$refs.license_link.href = licenseUrl(this.$props.value.shortName)
    },
    watch: {
        shortLicenseName: function(newVal, oldVal) {
            console.log(newVal, oldVal, this.$refs)
            // this.$refs.license_link.href = licenseUrl(this.$props.value.shortName)
        }
    }
}
</script>
<style>
    .chooser-selected {
        border-radius: 5px;
    }

    .chooser-selected h2 {
        margin: 1.5% auto;
        text-align: center;
    }
    #SelectedLicenseInfo h4 {
        line-height: 2;
        font-weight: 600;
        margin-top: 1rem;
        margin-bottom: 0.5rem;
    }

    .chooser-selected img {
        display: inline;
        vertical-align: top;
    }

    .selected-license-names b {
        font-size: 1.8rem;
    }
    .license-icons svg {
        width: 25px!important;
        height: 25px!important;
    }
    .switch-container p, b-switch {
        display: inline;
    }
    .readable-string {
        vertical-align: top;
        padding-left: 20px;
        font-weight: 500;
    }
</style>
