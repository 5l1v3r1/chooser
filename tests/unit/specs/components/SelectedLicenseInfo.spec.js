import { shallowMount, createLocalVue } from '@vue/test-utils'
import SelectedLicenseInfo from '@/components/SelectedLicenseInfo'
import Buefy from 'buefy'

describe('SelectedLicenseInfo.vue', () => {
    let wrapper

    // Always creates a shallow instance of component
    beforeEach(() => {
        const localVue = createLocalVue()
        localVue.use(Buefy)

        wrapper = shallowMount(SelectedLicenseInfo, {
            localVue,
            propsData: {
                value: {
                    shortName: 'CC BY 4.0',
                    fullName: 'Attribution 4.0 International'
                }
            }
        })
    })

    // Snapshot tests
    it('has the expected UI', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
