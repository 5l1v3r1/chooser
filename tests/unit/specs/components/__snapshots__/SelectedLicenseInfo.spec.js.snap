// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedLicenseInfo.vue has the expected UI 1`] = `
<div class="selected-license-info">
  <h4><a href="https://creativecommons.org/licenses/by/4.0/?ref=ccchooser">Attribution 4.0 International</a></h4>
  <p id="chooser-selected-description"><b>CC BY </b> <span></span></p>
  <section class="license-visual-info">
    <ul class="license-list"><span name="highlight"><li><licenseiconography-stub iconlist="by"></licenseiconography-stub> <span class="readable-string">
                        
                    </span></li></span></ul>
    <div class="freeworks-status__div"><a href="https://creativecommons.org/freeworks"><img src="../assets/license-icons/fc_approved_small.png" alt=""></a></div>
    <p class="freeworks-status__p"><a href="https://creativecommons.org/freeworks">
      </a></p>
  </section>
</div>
`;
